### PR TITLE
dasharo-deploy: update smmstore and bootsplash migration console output

### PR DESCRIPTION
**Pull request by mkopec**
Make the console output for settings and logo migration a bit more understandable to users, and pipe all cbfstool messages including stderr to error logs

### CODE REVIEW

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 1]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
+    if [ $? -ne 0 ]; then
+        echo "Failed! Default settings will be used."
+        return 1
+    fi
+    cbfstool "$BIOS_UPDATE_FILE" write -r SMMSTORE -f /tmp/smmstore.bin -u >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
+        echo "Failed! Default settings will be used."
+        return 1
+    fi
+    echo Done.
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: done: https://github.com/Dasharo/meta-dts/pull/76/commits/7ae66ff3a426793ff2d897e2f37a5982819dab24

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 2]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
     fi
-    echo "Migrate logo.bmp to BOOTSPLASH section."
+    echo -n "Backing up custom boot logo... "
     cbfstool $BIOS_UPDATE_FILE remove -n logo.bmp -r BOOTSPLASH >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
+        echo "Failed! Default boot splash will be used."
+        return 1
+    fi
     cbfstool $BIOS_UPDATE_FILE add -f /tmp/logo.bmp -r BOOTSPLASH -n logo.bmp -t raw -c lzma >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: done: https://github.com/Dasharo/meta-dts/pull/76/commits/7ae66ff3a426793ff2d897e2f37a5982819dab24 , https://github.com/Dasharo/meta-dts/pull/76/commits/b651fc257986c8788c68d5a287f132d33a040d0b

needed to ensure error_check doesn't print an empty line: https://github.com/Dasharo/meta-dts/pull/76/commits/f8abcd91136b90041636a092675e4d728b4bdbb7

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 3]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
 }
 
 bootsplash_migration() {
-    echo "Beginning BOOTSPLASH migration process..."
     flashrom -p "$PROGRAMMER_BIOS" ${FLASH_CHIP_SELECT} -r /tmp/dasharo_dump.rom ${FLASHROM_ADD_OPT_READ} --fmap -i FMAP -i BOOTSPLASH >> $FLASHROM_LOG_FILE 2>> $ERR_LOG_FILE
     cbfstool /tmp/dasharo_dump.rom extract -r BOOTSPLASH -n logo.bmp -f /tmp/logo.bmp >> $ERR_LOG_FILE 2>&1
     if [ $? -ne 0 ]; then
-      echo "logo.bmp not found, nothing to migrate"
-      return 1
+        # No logo, so exit early and don't show unnecessary messages
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: needed to ensure error_check doesn't print an empty line: https://github.com/Dasharo/meta-dts/commit/f8abcd91136b90041636a092675e4d728b4bdbb7

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 4]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
+        return 1
     fi
-    echo "Migrate logo.bmp to BOOTSPLASH section."
+    echo -n "Backing up custom boot logo... "
     cbfstool $BIOS_UPDATE_FILE remove -n logo.bmp -r BOOTSPLASH >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: done: https://github.com/Dasharo/meta-dts/commit/7ae66ff3a426793ff2d897e2f37a5982819dab24

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 5]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
-    cbfstool /tmp/dasharo_dump.rom read -r SMMSTORE -f /tmp/smmstore.bin
-    echo "Migrate to SMMSTORE section."
-    cbfstool "$BIOS_UPDATE_FILE" write -r SMMSTORE -f /tmp/smmstore.bin -u
+    cbfstool /tmp/dasharo_dump.rom read -r SMMSTORE -f /tmp/smmstore.bin >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
+        echo "Failed! Default settings will be used."
+        return 1
+    fi
+    cbfstool "$BIOS_UPDATE_FILE" write -r SMMSTORE -f /tmp/smmstore.bin -u >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: done: https://github.com/Dasharo/meta-dts/commit/7ae66ff3a426793ff2d897e2f37a5982819dab24

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 6]
```diff
@@ -226,24 +226,40 @@ smbios_migration() {
...
 
 smmstore_migration() {
-    echo "Beginning SMMSTORE migration process..."
+    echo -n "Backing up firmware configuration... "
     flashrom -p "$PROGRAMMER_BIOS" ${FLASH_CHIP_SELECT} -r /tmp/dasharo_dump.rom ${FLASHROM_ADD_OPT_READ} --fmap -i FMAP -i SMMSTORE >> $FLASHROM_LOG_FILE 2>> $ERR_LOG_FILE
-    cbfstool /tmp/dasharo_dump.rom read -r SMMSTORE -f /tmp/smmstore.bin
-    echo "Migrate to SMMSTORE section."
-    cbfstool "$BIOS_UPDATE_FILE" write -r SMMSTORE -f /tmp/smmstore.bin -u
+    cbfstool /tmp/dasharo_dump.rom read -r SMMSTORE -f /tmp/smmstore.bin >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
```
[the remark this answers is not in the document]
Comment (mkopec) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: done: https://github.com/Dasharo/meta-dts/commit/7ae66ff3a426793ff2d897e2f37a5982819dab24

**Comment by mkopec** [comment]
@TomaszAIR Addressed your review, but I've not tested any of these changes yet

**Comment by mkopec** [comment]
@TomaszAIR ping :) I don't think I'll be adding any more UX changes in this PR

**Comment by TomaszAIR** [comment]
@mkopec After tests I spotted that using everywhere just error_checks may lead to exit from whole script in cases where e.g. we cannot migrate boot logo.

I pushed couple modifications, could you please review? With this I got following outcome while updating nv4x adl

```
Enter an option: 5

Waiting for network connection ...
Checking if board is Dasharo compatible.
Gathering flash chip and chipset information...
Flash information: vendor="Programmer" name="Opaque flash chip"
Flash size: 32M
Getting platform specific GPG key... 
Done
AC adapter is connected. Continuing with firmware update.
Checking for the latest Dasharo update available...
Current Dasharo version: 1.7.1
Latest available Dasharo version: 1.7.2
Downloading Dasharo firmware...

Please verify detected hardware!

Board vendor: Notebook
System model: NV4xPZ
Board model: NV4xPZ

Does it match your actual specification? (Y|n) y

Following firmware will be used to install Dasharo
Dasharo BIOS firmware:
  - link: https://3mdeb.com/open-source-firmware/Dasharo/novacustom_nv4x_adl/v1.7.2/novacustom_nv4x_adl_v1.7.2.rom
  - hash: 00b6338389cc5d020b641629971aac6d4047be6134c6e8d0228140edc42584f6
Dasharo EC firmware:
  - link: https://3mdeb.com/open-source-firmware/Dasharo/novacustom_nv4x_adl/v1.7.2/novacustom_nv4x_adl_ec_v1.7.2.rom
  - hash: adbd651646789f5ac4c9d1835f0d2bf502192fd9476cd77b9b22fa077c5403e8

You can learn more about this release on: https://docs.dasharo.com/

Do you want to update Dasharo firmware on your hardware? (Y|n) y

Checking Dasharo firmware checksum...
Done
Checking Dasharo firmware signature...
gpg: Signature made Fri Dec 22 16:36:15 2023 UTC
gpg:                using RSA key A5D32D10A7046F99F718908377BB8801DAB4EDCF
gpg: Good signature from "Novacustom Open Source Firmware Release 1.x Signing Key" [full]
Done
Backing up firmware configuration... 
Done.
Updating Dasharo firmware...
This will take around 3 minutes. Please be patient and do not reset your computer, or touch the keyboard!
Dasharo EC update process will start in a moment.
Your machine will shut down automatically once it finishes.
This will take around 1 minute. Please be patient and do not reset your computer, or touch keyboard!
Updating Dasharo EC firmware...
Checking Dasharo EC firmware checksum...
Done
Checking Dasharo EC firmware signature...
gpg: Signature made Fri Dec 22 16:36:13 2023 UTC
gpg:                using RSA key A5D32D10A7046F99F718908377BB8801DAB4EDCF
gpg: Good signature from "Novacustom Open Source Firmware Release 1.x Signing Key" [full]
Done
Updating EC...
```

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 7]
```diff
@@ -226,24 +226,26 @@ smbios_migration() {
 }
 
 smmstore_migration() {
-    echo "Beginning SMMSTORE migration process..."
+    echo "Backing up firmware configuration... "
```
Comment (mkopec) [inline meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy:229]:
```suggestion
    echo -n "Backing up firmware configuration... "
```

I wanted to have the Done or Error message printed in the same line to make it a bit shorter and nicer to read

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy [hunk 8]
```diff
@@ -226,24 +226,26 @@ smbios_migration() {
...
-      echo "logo.bmp not found, nothing to migrate"
-      return 1
-    fi
-    echo "Migrate logo.bmp to BOOTSPLASH section."
-    cbfstool $BIOS_UPDATE_FILE remove -n logo.bmp -r BOOTSPLASH >> $ERR_LOG_FILE 2>&1
-    cbfstool $BIOS_UPDATE_FILE add -f /tmp/logo.bmp -r BOOTSPLASH -n logo.bmp -t raw -c lzma >> $ERR_LOG_FILE 2>&1
+    # If no custom logo, return from bootsplash_migration early and don't show
+    # unnecessary messages
+    cbfstool /tmp/dasharo_dump.rom extract -r BOOTSPLASH -n logo.bmp -f /tmp/logo.bmp >> $ERR_LOG_FILE 2>&1 || return 1
+    echo "Backing up custom boot logo... "
```
Comment (mkopec) [inline meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy:243]:
```suggestion
    echo -n "Backing up custom boot logo... "
```

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh [hunk 9]
```diff
@@ -702,8 +705,10 @@ download_keys() {
 }
 
 get_signing_keys() {
-    wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$PLATFORM_SIGN_KEY -O - | gpg --import -
+    echo "Getting platform specific GPG key... "
```
Comment (mkopec) [inline meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh:708]:
```suggestion
    echo -n "Getting platform specific GPG key... "
```

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy (hunk 7 above)
Comment (TomaszAIR) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: @mkopec got it, added in https://github.com/Dasharo/meta-dts/pull/76/commits/dafd803a5b67db2049ff76ea7a746d62caeb4acc

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh (hunk 9 above)
Comment (TomaszAIR) [reply meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh]: Added in https://github.com/Dasharo/meta-dts/pull/76/commits/dafd803a5b67db2049ff76ea7a746d62caeb4acc

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh [hunk 10]
```diff
@@ -729,11 +734,13 @@ verify_artifacts() {
     ;;
   esac
   echo "Checking $_name firmware checksum..."
```
[the remark this answers is not in the document]
Comment (TomaszAIR) [reply meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh]: Added in https://github.com/Dasharo/meta-dts/pull/76/commits/dafd803a5b67db2049ff76ea7a746d62caeb4acc

File: meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy (hunk 8 above)
Comment (TomaszAIR) [reply meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy]: Added in https://github.com/Dasharo/meta-dts/pull/76/commits/dafd803a5b67db2049ff76ea7a746d62caeb4acc

**Comment by TomaszAIR** [comment]
Merging

**Pull request merged**